x-pack/plugins/lens/public/xy_visualization/expression.tsx

```diff
@@ -387,6 +430,16 @@ export function XYChart({
     return style;
   };
 
+  const shouldShowValueLabels =
+    // No stacked bar charts
+    filteredLayers.every((layer) => !layer.seriesType.includes('stacked')) &&
+    // No histogram charts
+    !isHistogramViz;
+
+  const baseThemeWithMaybeValueLabels = !shouldShowValueLabels
+    ? chartTheme
+    : mergeThemeWithValueLabelsStyling(chartTheme, valueLabels || 'hide', shouldRotate);
```

Comment:
nit: `valueLabels` should always be set, right?

Reply:
I think so. I was thinking about backward compatibility here, but probably the expression evaluation should ensure that the value is set.

Reply:
Yep, expressions are regenerated every time, so it's safe to assume it's available here because of the default value in `to_expression`